src/test/java/org/jitsi/meet/test/pageobjects/web/WebRemoteParticipant.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright @ 2018 8x8 Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.meet.test.pageobjects.web;
+
+import org.jitsi.meet.test.pageobjects.base.*;
+import org.jitsi.meet.test.util.*;
+import org.openqa.selenium.*;
+import org.openqa.selenium.interactions.*;
+
+/**
+ * The web implementation of <tt>RemoteParticipant/tt>.
+ *
+ * @author Damian Minkov
+ */
+public class WebRemoteParticipant
+    extends RemoteParticipant<WebDriver>
+{
+    /**
+     * Constructs <tt>WebRemoteParticipant</tt>.
+     * @param driver the driver.
+     * @param endpointId the endpoint id.
+     */
+    public WebRemoteParticipant(WebDriver driver, String endpointId)
+    {
+        super(driver, endpointId);
+    }
+
+    @Override
+    public void mute()
+    {
+        clickOnRemoteMenuLink("mutelink");
+    }
+
+    @Override
+    public void kick()
+    {
+        clickOnRemoteMenuLink("kicklink");
+    }
+
+    /**
+     * Opens the remote menu and waits for the desired link to appear.
+     * @param linkClassname the class name to use identifying the link.
+     */
+    private void clickOnRemoteMenuLink(String linkClassname)
+    {
+        // Open the remote video menu
+        WebElement cntElem
+            = driver.findElement(By.id("participant_" + this.getEndpointId()));
+        String remoteVideoMenuButtonXPath
+            = TestUtils.getXPathStringForClassName(
```

Comment:
Usage of this helper is off to me, considering this might be the only place it gets used, meaning it has inconsistent usage so I think we should use it more or kill it. You can probably kill this helper by changing what By you're using. Without changing jitsi-meet more, you can just look for By.classname (or maybe use an xpath but check it includes the class). Ideally you try to keep selectors as concise as possible so that when the dom/layout changes the tests don't have to be modified, and tying to a span adds one more reason to change.

Reply:
I will look at it, tomorrow to killing it.